tvb_multiscale/core/tvb/io/h5_writer.py

```diff
@@ -152,28 +175,32 @@ def _prepare_object_for_group(self, group, object, h5_type_attribute="", nr_regi
                 return group, subgroups
 
     def write_object(self, object, h5_type_attribute="", nr_regions=None,
-                     path=None, h5_file=None, close_file=True):
+                     path=None, close_file=True):
         """
                 :param object: object to write recursively in H5
                 :param path: H5 path to be written
         """
-        h5_file, path = self._open_file(object.__class__.__name__, path, h5_file)
-        h5_file = self._prepare_object_for_group(h5_file, object, h5_type_attribute, nr_regions)
-        self._close_file(h5_file, close_file)
-        self._log_success(object.__class__.__name__, path)
-        return h5_file, path
+        self._assert_file(path, object.__class__.__name__)
+        try:
+            self.hfd5_target = self._prepare_object_for_group(self.hfd5_target, object, h5_type_attribute, nr_regions)
+            self._log_success_or_warn(None, object.__class__.__name__)
+        except Exception as e:
+            self._log_success_or_warn(e, object.__class__.__name__)
+        self._close_file(close_file)
 
-    def write_list_of_objects(self, list_of_objects, path=None, h5_file=None, close_file=True):
-        h5_file, path = self._open_file("List of objects", path, h5_file)
-        for idict, object in enumerate(list_of_objects):
-            idict_str = str(idict)
-            h5_file.create_group(idict_str)
-            self.write_object(object, h5_file=h5_file[idict_str], close_file=False)
-        h5_file.attrs.create(self.H5_TYPE_ATTRIBUTE, numpy.string_("List of objects"))
-        h5_file.attrs.create(self.H5_SUBTYPE_ATTRIBUTE, numpy.string_("list"))
-        self._close_file(h5_file, close_file)
-        self._log_success("List of objects", path)
-        return h5_file, path
+    def write_list_of_objects(self, list_of_objects, path=None, close_file=True):
+        self._assert_file(path, "List of objects")
+        try:
+            for idict, object in enumerate(list_of_objects):
+                idict_str = str(idict)
+                h5_file.create_group(idict_str)
```

Comment:
@dionperd there seem some errors related to 'h5_file' being undefined. Did you intend to declare it as a parameter in the method's signature? For example something like:
`def write_list_of_objects(self, list_of_objects, path=None, close_file=True, h5_file=None):`

Reply:
Hello, no it should become self.hfd5_target like above!
Thanks for finding this out!